lib/system.rb

```diff
@@ -0,0 +1,74 @@
+require_relative 'room.rb'
+require_relative 'reservation.rb'
+require_relative 'reservation_date.rb'
+
+module Hotel
+  class System
+    attr_reader :rooms, :reservations, :start_date, :end_date
+    
+    def initialize room_num: nil, start_date: nil, end_date: nil, rate: nil
+      @rooms = Hotel::Room.all
+      @reservations = Array.new
+      @new_reservation = nil #temporary varialbe for booking
+    end
+    
+    def room_list
+      return Hotel::Room.all
+    end
+    
+    def create_reservation(room_num: nil, start_date: start_date, end_date: end_date, rate: nil)
+      #reservation_validation(start_date, end_date)
+      Hotel::Reservation.new(start_date: start_date, end_date: end_date, rate: 200)
+    end
+    
+    def add_reservation(new_reservation)
+      @reservations << new_reservation
+    end
+    
+    def reservation_validation(start_date, end_date)
```

Comment:
This looks like a useful method, and makes good use of the `overlap_check` method from `Reservation_Date`. You are on the right track.